src/HDInsight/HDInsight/ManagementCommands/SetAzureHDInsightGatewayCredentialCommand.cs

```diff
@@ -0,0 +1,151 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Hyak.Common;
+using Microsoft.Azure.Commands.HDInsight.Commands;
+using Microsoft.Azure.Commands.HDInsight.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.HDInsight.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.WindowsAzure.Commands.Common;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.HDInsight
+{
+    [Cmdlet(VerbsCommon.Set, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "HDInsightGatewayCredential", DefaultParameterSetName = SetByNameParameterSet, SupportsShouldProcess = true), OutputType(typeof(HttpConnectivitySettings))]
+    public class SetAzureHDInsightGatewayCredentialCommand : HDInsightCmdletBase
+    {
+        private const string SetByNameParameterSet = "SetByNameParameterSet";
+        private const string SetByResourceIdParameterSet = "SetByResourceIdParameterSet";
+        private const string SetByInputObjectParameterSet = "SetByInputObjectParameterSet";
+
+        #region Input Parameter Definitions
+
+        [Alias("ClusterName")]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = SetByNameParameterSet,
+            HelpMessage = "Gets or sets the name of the cluster.")]
+        [ResourceNameCompleter("Microsoft.HDInsight/clusters", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string Name
+        {
+            get { return _clusterName; }
+            set { _clusterName = value; }
+        }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = SetByInputObjectParameterSet,
+            HelpMessage = "Gets or sets the input object.")]
+        [ValidateNotNull]
+        public AzureHDInsightCluster InputObject;
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = SetByResourceIdParameterSet,
+            HelpMessage = "Gets or sets the resource id.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId;
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = SetByNameParameterSet,
+            HelpMessage = "Gets or sets the login for the cluster's user.")]
+        [Parameter(
```

Comment:
Note that if a parameter is in every parameter set, you only need to specify a single parameter attribute, and not specify the ParameterSetName property (this puts the parameter in all parameter sets).

This is OK - Just FYI for the future

Reply:
> Note that if a parameter is in every parameter set, you only need to specify a single parameter attribute, and not specify the ParameterSetName property (this puts the parameter in all parameter sets).
> 
> This is OK - Just FYI for the future

Fixed. Thanks for your review.